### ISSUE TITLE
spotSources in grid views

### ISSUE THREAD

**Issue by constantinpape** [mobie/spatial-transcriptomics-example-project#17]
The discussion in https://github.com/mobie/mobie-viewer-fiji/issues/827 made me realize that we will actually need to be able to put spotSources into grid views as well.
Does this currently work in your concepts, @tischi? In particular does it work for a mergedGrid, where you would need to know the source dimensions etc.? Could this be solved by using the `metadataSource` field and referencing an image source with the correct dimensions?

**Comment by tischi** [comment]
This should just work, I create a label mask image from the spots table like this:

```Java
kdTree = new KDTree( new ArrayList<>( annotatedSpots ), new ArrayList<>( annotatedSpots ) );
mask = GeomMasks.closedBox( kdTree.minAsDoubleArray(), kdTree.maxAsDoubleArray() );
final Interval interval = Intervals.smallestContainingInterval( getMask() );
final FunctionRealRandomAccessible< UnsignedIntType > realRandomAccessible = new FunctionRealRandomAccessible( 3, new LocationToLabelSupplier(), UnsignedIntType::new );
source = new RealRandomAccessibleIntervalTimelapseSource<>( realRandomAccessible, interval, new UnsignedIntType(), new AffineTransform3D(), name, true, timePoints );
```

TL;DR: the image dimensions are constructed from the min and max spot locations.

**Comment by constantinpape** [comment]
> TL;DR: the image dimensions are constructed from the min and max spot locations.

But that could yield the wrong result, because the spots are sparse.
If you have `min(spots) = 1.0, max(spots) = 246`, but `image shape = 256` it could lead to misalignment.

**Comment by tischi** [comment]
where would `image shape = 256` come from?

**Comment by constantinpape** [comment]
> where would `image shape = 256` come from?

In the case where the spots are associated with a particular image.
Easiest example would be a grid, where one source per position is an image and the other source per position are spots.

**Comment by tischi** [comment]
> Easiest example would be a grid, where one source per position is an image and the other source per position are spots.

For `mergedGrid` this would mean making two grids and for the one with the spots the `metadataSource` concept would come in very handy!

For `transformedGrid`... we don't have `metadataSource` there, and I am not sure how to do this...

Hmmm, this feels like the `annotation` layer that I proposed in the context of the `regions` would be something, because there would could explicitly create a `spotsImage` from a `spotsTable` and some metadata...

Maybe we add `realDimensions` (not sure who to call it) as an optional field to the `spots` `source`? Or the `spots` source could refer to another `imageSource` for its metadata? Con: we would introduce that sources depend on each other and are not stand-alone...

**Comment by constantinpape** [comment]
> Maybe we add `realDimensions` (not sure who to call it) as an optional field to the `spots` `source`?

Yes, I think that would be my preferred solution. I guess we can call it `pixelDimensions` (with the understanding that this can be 2 and 3d depending on the data dimensionality)?

**Comment by tischi** [comment]
But we need the "full extend", not the size of one pixel....

**Comment by constantinpape** [comment]
> But we need the "full extend", not the size of one pixel....

Yeah, that's what I mean with this, but if the name is not clear to you, than maybe: `extentInPixels`?

**Comment by constantinpape** [comment]
Oh, and do you actually need the extent in pixels or the physical extent here?

**Comment by constantinpape** [comment]
Given that you compute this from the spot coordinates I guess you need the physical extent. So I would suggest either `physicalExtent` or `extent` as the name. (If you prefer we can also call it dimensions instead of extent).

**Comment by constantinpape** [comment]
@tischi I have added `boundingBoxMin`, `boundingBoxMax` and `unit` as discussed. Also, I made these fields mandatory now, so you can rely on them being there; I think the information should always be given.
